.github/workflows/build-test.yml

```diff
@@ -125,27 +127,66 @@ jobs:
           artifact-bin-name: cargo-nextest
           artifact-upload-name: ${{ runner.os }}-cargo-nextest
 
-      - name: Prepare system
-        run: ${{ matrix.os == 'windows-latest' && 'internal\scripts\ci_prepare_windows.bat' || ( matrix.os == 'ubuntu-latest' && './internal/scripts/ci_prepare_ubuntu.sh' || 'uname -a' ) }}
+      - name: Prepare Windows
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: internal\scripts\ci_prepare_windows.bat
+      - name: Prepare Linux
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: |
+          internal/scripts/ci_prepare_ubuntu.sh
+          uname -a
 
       - name: Run cargo build
         run: cargo build --workspace --all-targets ${{ matrix.mode.arg }} --target i686-unknown-linux-gnu
 
       - name: Run cargo nextest
         run: cargo nextest run --workspace --no-fail-fast ${{ matrix.mode.arg }} --target i686-unknown-linux-gnu
 
+      - name: Build language bindings
+        run: |
+          cmake . \
+                -B target/ffi/build \
+                -DCMAKE_INSTALL_PREFIX=target/ffi/install \
+                -DEXAMPLES=ON \
+                -DRUST_TARGET_TRIPLET="i686-unknown-linux-gnu" \
+                -DCMAKE_C_FLAGS="-m32" \
+                ${{ matrix.mode.cmake-build-type }}
+          cmake --build target/ffi/build
+          cmake --install target/ffi/build
+
+      - name: Build language binding examples in out-of-tree configuration
+        if: false # out-of-tree cross-compilation is currently not supported
+        run: |
+          rm -rf target/ffi/build
+          cmake examples/c \
+                -B target/ffi/out-of-tree \
+                -DCMAKE_PREFIX_PATH=${{ github.workspace }}/target/ffi/install \
+                -DCMAKE_C_FLAGS="-m32" \
+                -${{ matrix.mode.cmake-build-type }}
+          cmake --build target/ffi/out-of-tree
+
   x86_64:
     needs: [preflight-check, static-code-analysis, cargo-nextest]
     if: ${{ needs.changes.outputs.source-code == 'true' }}
     strategy:
       matrix:
         os: [windows-latest, ubuntu-latest, macos-latest]
-        toolchain: [stable, 1.75.0]
-        mode: 
+        toolchain: [stable, stable-gnu, 1.75.0]
+        mode:
           - name: "release"
             arg: "--release"
+            cmake-build-type: "-DCMAKE_BUILD_TYPE=Release"
           - name: "debug"
             arg: ""
+            cmake-build-type: "-DCMAKE_BUILD_TYPE=Debug"
+        include:
+          - toolchain: stable-gnu
+            cmake-build-system-generator: '-G "MinGW Makefiles"'
+        exclude:
+          - os: ubuntu-latest
+            toolchain: stable-gnu
+          - os: macos-latest
+            toolchain: stable-gnu
```

Comment:
I think I could merge the 32 bit build to this matrix build in a similar fashion, e.g. by adding a `bits: [64, 32]`, excluding the 32 bit builds on anything but ubuntu with the stable toolchain and including the target-triplet for 32 bits. What do you think? Will this become too complex?

Reply:
I would create a matrix for every architecture. On the other hand, it  is interesting to have a big matrix for all OS, all architectures, and all rust versions - but this can soon be too much in the form of CI targets and complexity in this file.

So I would move the 32 bit tests into the x86_32 matrix.